File: StriverA2Z/JavaScript&TypeScript/src/BinaryTrees/heightOfBT.ts
import {arrayToBTLevelOrder, BTNode} from "./BinaryTreeOperations";

function heightOfBT(root: BTNode) {
	if (root === null) return 0;
	const l = heightOfBT(root.left);
	const r = heightOfBT(root.right);
	return 1 + Math.max(l, r);
}

let root = arrayToBTLevelOrder([1,2,3,4,5,6,7,8,9,10,11]);
console.log(heightOfBT(root));

File: StriverA2Z/C++/13_BinaryTree/height_of_BT.cpp
#include<bits/stdc++.h>
#include "Node.h"

using namespace std;

int heightOfBT(Node *root) {
    if (root == nullptr) return 0;
    int left = heightOfBT(root->left);
    int right = heightOfBT(root->right);
    return 1 + max(left, right);
}

int main() {
    Node *root = arrayToBTLevelOrder({1, 2, 3, 4, 5, 6, 7, 8, 9});
    int len = heightOfBT(root);
    cout << len;
    return 0;
}